dart/lib/src/utils/sample_rate_format.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:meta/meta.dart';
+
+@internal
+class SampleRateFormat {
+  static String format(double sampleRate) {
+    final fixed = sampleRate.toStringAsFixed(16);
+    return fixed.replaceAll(RegExp(r"([.]*0+)(?!.*\d)"), "");
```

Comment:
Are we using the same Regex from `NumberFormat`?

Reply:
NumberFormat does a bunch of things. This regex just removes trailing zeros from the string.

Reply:
But the `dp` does a bunch of things and I am unsure if this has the same result, can you reason around the `pow` and `round / mod`?